simulation/simple_sensor_simulator/test/src/sensor_simulation/primitives/test_primitive.hpp

```diff
@@ -0,0 +1,96 @@
+// Copyright 2015 TIER IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SIMPLE_SENSOR_SIMULATOR__TEST__TEST_PRIMITIVE_HPP_
+#define SIMPLE_SENSOR_SIMULATOR__TEST__TEST_PRIMITIVE_HPP_
+
+#include <gtest/gtest.h>
+#include <quaternion_operation/quaternion_operation.h>
+
+#include <geometry_msgs/msg/point.hpp>
+#include <geometry_msgs/msg/pose.hpp>
+#include <optional>
+#include <simple_sensor_simulator/sensor_simulation/primitives/primitive.hpp>
+#include <vector>
+
+using namespace simple_sensor_simulator::primitives;
+using namespace simple_sensor_simulator;
+
+namespace simple_sensor_simulator
+{
+namespace primitives
+{
+
+class DummyPrimitive : public Primitive
+{
+public:
+  DummyPrimitive(const std::string & type, const geometry_msgs::msg::Pose & pose)
+  : Primitive(type, pose)
+  {
+  }
+
+  auto setVertices(const std::vector<Vertex> & vertices) -> void { vertices_ = vertices; }
+  auto setTriangles(const std::vector<Triangle> & triangles) -> void { triangles_ = triangles; }
+
+  auto getVerticesSize() const -> int { return vertices_.size(); }
+  auto getTrianglesSize() const -> int { return triangles_.size(); }
+
+  auto getVertices() const -> const std::vector<Vertex> & { return vertices_; }
+  auto getTriangles() const -> const std::vector<Triangle> & { return triangles_; }
+};
+
+}  // namespace primitives
+}  // namespace simple_sensor_simulator
```

Comment:
Tier4 members have [refused](https://github.com/tier4/scenario_simulator_v2/pull/1268#issuecomment-2157121733) a PR containing a Dummy test class in the past. Unfortunately I'm not sure how getters and setters can be replaced in order to test these functionalities.

Reply:
I see, but i think i can try with this DummyPrimitive. In my opinion we have to test primitive instead of Box.